From 9ea8be71c090a58d4d15466799dd43883e965610 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 1 Jan 2019 21:57:57 +0100
Subject: [PATCH 4/4] GCC doesn't support cortex-a55/75, fallback to cortex-a53

* Instead of optimizing for a generic ARM/ARM64 cpu,
  optimize to the closest supported CPU supported by GCC
* Also removes some spurious build warnings while building
  toolchain libraries ( libatomic, libgcc... )

Test: Check Arm64CortexA55Cflags mcpu in soong/build.ninja
Test: m libc
Change-Id: I7db9632628ce487adf87b9e62ae929248fa931a2
---
 cc/config/arm64_device.go | 5 +++++
 cc/config/arm_device.go   | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/cc/config/arm64_device.go b/cc/config/arm64_device.go
index 8be33a85..992a88ed 100644
--- a/cc/config/arm64_device.go
+++ b/cc/config/arm64_device.go
@@ -94,6 +94,11 @@ func init() {
 	// Clang supports specific Kryo targeting
 	replaceFirst(arm64ClangCpuVariantCflags["kryo"], "-mcpu=cortex-a57", "-mcpu=kryo")
 
+	// cortex-a55 and cortex-a75 are not supported by GCC, but are supported by Clang,
+	// so override the definitions when building modules with Clang.
+	replaceFirst(arm64CpuVariantCflags["cortex-a55"], "-mcpu=cortex-a55", "-mcpu=cortex-a53")
+	replaceFirst(arm64CpuVariantCflags["cortex-a75"], "-mcpu=cortex-a55", "-mcpu=cortex-a53")
+
 	pctx.StaticVariable("arm64GccVersion", arm64GccVersion)
 
 	pctx.SourcePathVariable("Arm64GccRoot",
diff --git a/cc/config/arm_device.go b/cc/config/arm_device.go
index cad1b161..82f26f81 100644
--- a/cc/config/arm_device.go
+++ b/cc/config/arm_device.go
@@ -175,6 +175,11 @@ func init() {
 	// override the definitions when building modules with Clang.
 	replaceFirst(armClangCpuVariantCflags["krait"], "-mcpu=cortex-a15", "-mcpu=krait")
 
+	// cortex-a55 and cortex-a75 are not supported by GCC, but are supported by Clang,
+	// so override the definitions when building modules with Clang.
+	replaceFirst(armCpuVariantCflags["cortex-a55"], "-mcpu=cortex-a55", "-mcpu=cortex-a53")
+	replaceFirst(armCpuVariantCflags["cortex-a75"], "-mcpu=cortex-a55", "-mcpu=cortex-a53")
+
 	// The reason we use "-march=armv8-a+crc", instead of "-march=armv8-a", for
 	// gcc is the latter would conflict with any specified/supported -mcpu!
 	// All armv8-a cores supported by gcc 4.9 support crc, so it's safe
-- 
2.17.1

